src/ilcdlib/entity/compliance.py

```diff
@@ -0,0 +1,117 @@
+#
+#  This software was developed with support from the Skanska USA,
+#  Charles Pankow Foundation, Microsoft Sustainability Fund, Interface, MKA Foundation, and others.
+#  Find out more at www.BuildingTransparency.org
+#
+from typing import Type
+
+from openepd.model.standard import Standard
+
+from ilcdlib.common import (
+    BaseIlcdMediumSpecificReader,
+    IlcdXmlReader,
+    OpenEpdComplianceSupportReader,
+    OpenEpdStandardReader,
+)
+from ilcdlib.dto import IlcdReference
+from ilcdlib.type import LangDef
+from ilcdlib.utils import none_throws
+from ilcdlib.xml_parser import T_ET
+
+
+class IlcdStandardReader(OpenEpdStandardReader, IlcdXmlReader):
+    """Reader that can parser an ILCD Standard specification from an XML file."""
+
+    def __init__(
+        self,
+        element: T_ET.Element,
+        data_provider: BaseIlcdMediumSpecificReader,
+    ):
+        super().__init__(data_provider)
+        self._entity = element
+
+    def get_uuid(self) -> str:
+        """Get the UUID of the entity described by this data set."""
+        return none_throws(
+            self._get_text(self._entity, ("source:sourceInformation", "source:dataSetInformation", "common:UUID"))
+        )
+
+    def get_version(self) -> str | None:
+        """Get the version of the entity described by this data set."""
+        return self._get_text(
+            self._entity,
+            ("source:administrativeInformation", "source:publicationAndOwnership", "common:dataSetVersion"),
+        )
+
+    def get_own_reference(self) -> IlcdReference | None:
+        """Get the reference to this data set."""
+        return IlcdReference(entity_type="sources", entity_id=self.get_uuid(), entity_version=self.get_version())
+
+    def get_short_name(self) -> str:
+        """Get the short name of the entity described by this data set."""
+        short_name = self._get_text(
+            self._entity,
+            ("source:sourceInformation", "source:dataSetInformation", "common:shortName"),
+        )
+        return short_name or f"Name is missing. [Standard {self.get_uuid()}]"
+
+    def get_name(self) -> str | None:
+        """Get the full name of the entity described by this data set."""
+        return self._get_text(
+            self._entity,
+            ("source:sourceInformation", "source:dataSetInformation", "source:sourceDescriptionOrComment"),
+        )
+
+    def to_openepd_standard(self, lang: LangDef, base_url: str | None = None) -> Standard:
+        """Read a single OpenEPD Standard object."""
+        return Standard(
+            short_name=self.get_short_name(),
+            name=self.get_name(),
+            # link=None, not yet found examples'
+            # issuer=None, not yet found examples
+            # attachments=create_openepd_attachments(self.get_own_reference(), base_url)
+            # need to update openepd specification
+        )
+
+
+class IlcdComplianceReader(OpenEpdComplianceSupportReader, IlcdXmlReader):
```

Comment:
I'm not sure it is necessary to create a dedicated reader for that because it doesn't seem to have any business logic other just a simple iteration by elements. Instead it might be better just create `get_compliance_declarations` which will return a DTO containing compliance reader and `approvalOfOverallCompliance` (see ilcdlib.entity.flow.IlcdExchangeDto as an example).